### DISCUSSION TITLE
I think there might be some issues with Toggle Buttons, but I'm not sure.

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15565] [Q&A]
I have a group of Toggle Buttons, like this:

![2024-04-30 170026](https://github.com/AvaloniaUI/Avalonia/assets/9688223/bef5bb73-d763-41bb-b1dd-bfa2eae291fa)

And I've defined their values in my view model something like this:
`ObservableCollection<bool>  [true, false, false, false, false, false, false, false, false, false]`

And I'm binding them like this:
`<ToggleButton IsChecked="{Binding Selected[4]}">`

But when I click on the Toggle Buttons to switch their values, I get this error:
![2024-04-30 170414](https://github.com/AvaloniaUI/Avalonia/assets/9688223/a359614c-f071-48d4-9ddf-dac070a2772c)
and this error:
`[Binding]Error in binding to 'Avalonia.Controls.Primitives.ToggleButton'.'IsChecked': 'Could not convert 'System.Collections.ObjectModel.ObservableCollection`1[System.Boolean]' to 'Nullable`1'.'(ToggleButton #24013807)
[Binding]Error in binding to 'Avalonia.Controls.Primitives.ToggleButton'.'IsChecked': 'Could not convert 'System.Collections.ObjectModel.ObservableCollection`1[System.Boolean]' to 'Nullable`1'.'(ToggleButton #66070243)`

Am I using them incorrectly? Where did the System.Int32 value come from? Hi @maxkatz6

**Comment by timunie** [comment]
I don't suggest to do it like this. Better to have an ItemViewModel:

```cs
public class MyCheckItem : ViewModelBase
{
    public bool IsChecked {get; set;} // make sure to raise property changed, only dummy here!
}

ObservableCollection<MyCheckItem> Selected {get; } = [ new MyCheckItem(){IsCheched = treu } ];
```

```xml
<ToggleButton IsChecked="{Binding Selected[4].IsChecked}">
```
